### PR TITLE
Added gaps at the end of the storage of each contract.

### LINKED ISSUE #13: Consider leaving empty storage space in contracts for future enhancements
Adding new storage variables to a contract that has been extended by another is impossible, since the storage from the child contract will clash with the new variables added by the parent. To mitigate this, we can "reserve" a few slots in the base contracts, to allow for new variables to be added later. This could be done by declaring some fixed-length `uint256` arrays in storage, until https://github.com/ethereum/solidity/issues/4519 is implemented.

### PR DESCRIPTION
**Pull request by nventuro**
Fixes #13.

Despite `IERC721` and friends being `contracts` and not `interfaces` (due to a Solidity limitation), I kept this out of those, since they should be used as interfaces.

### CODE REVIEW

File: contracts/Initializable.sol [hunk 1]
```diff
@@ -46,4 +46,6 @@ contract Initializable {
     assembly { cs := extcodesize(address) }
     return cs == 0;
   }
+
+  uint256[50] private ______gap;
```
Comment (spalladino) [inline contracts/Initializable.sol]: We will be removing Initializable and loading it from zos-lib afterwards, right?

**Comment by spalladino** [comment]
I'd say we are fine with 50 slots per contract :-)

File: contracts/Initializable.sol (hunk 1 above)
Comment (nventuro) [reply contracts/Initializable.sol]: Yes - should we add this there?

**Comment by spalladino** [comment]
I'd say the opposite, better keep Initializable as clean as possible

**Comment by nventuro** [comment]
Any specific reason to not add the gap? I wouldn't discount the possibility of `Initializable` becoming more complex and needing more storage.

**Comment by spalladino** [comment]
I was going to suggest backwards compatibility with the Initializable from 1.x, but that one used a single boolean variable, while the current one uses two. And this exactly proves your point. So, can you send a PR to zeppelinos to add the gap there as well?

**Comment by nventuro** [comment]
Sure thing! Opened https://github.com/zeppelinos/zos/pull/215

**Pull request merged**